src/python/bot/fuzzers/syzkaller/engine.py

```diff
@@ -93,6 +99,27 @@ def _create_temp_corpus_dir(self, name):
     engine_common.recreate_directory(new_corpus_directory)
     return new_corpus_directory
 
+  def get_phone_family(self):
+    fingerprint = settings.get_build_fingerprint()
+    matcher = re.search(FINGERPRINT_REGEX, fingerprint)
+    return matcher.group(1)
+
+  def save_corpus(self, source, destination):
+    source_file = os.path.join(source, CORPUS_FILENAME)
+    if not os.path.exists(destination):
+      os.mkdir(destination)
+    target_file = os.path.join(destination, self.get_phone_family() + '.db')
+    if os.path.isfile(source_file):
+      shutil.copy(source_file, target_file)
+
+  def init_corpus(self, source, destination):
+    source_file = os.path.join(source, self.get_phone_family() + '.db')
+    if not os.path.exists(destination):
+      os.mkdir(destination)
+    destination_file = os.path.join(destination, CORPUS_FILENAME)
+    if os.path.isfile(source_file):
```

Comment:
we need to add
if os.path.isfile(source_file) and (not os.path.exists(destination_file) or os.path.getsize(source_file) > os.path.getsize(destination_file)):

this ensures that we only update corpus if it is updated.

Reply:
Done.